viamonvif/service.go

```diff
@@ -88,6 +107,9 @@ func newDiscovery(_ context.Context, _ resource.Dependencies,
 
 // DiscoverResources discovers different rtsp cameras that use onvif.
 func (dis *rtspDiscovery) DiscoverResources(ctx context.Context, extra map[string]any) ([]resource.Config, error) {
+	dis.mu.Lock()
+	defer dis.mu.Unlock()
```

Comment:
we don't need to lock for the entire `DiscoverResources` we only need to lock when we are updating `rtspToSnapshotURIs`
```suggestion
```

rewrite this to collect the `map[string]string{}` here https://github.com/viam-modules/viamrtsp/pull/122/files#diff-570f7bbcfc5375cfac6f0a2b08f355feca46353fd1b26dcde074981574d4ae4fR129

and only re assign it to `dis.RTSPToSnapshotURIs` at the end of the function under the mutex. 

This will reduce the likelyhood of many calls to DiscoverResources blocking behind the mutex unnecessarily.